### PR TITLE
Remove unneccsary template for user data source

### PR DESCRIPTION
**Pull request by drewmullen**
~~I believe its possible to provide default templates for resources and data sources that would automatically include examples (if found). This reduces the barrier for new contributors, peer reviewers and ensures a better and more consistent doc experience~~

Update - i thought i was required to set a template but those are optional. initially i create this PR to add default templates but, turns out there is a default template. When i initially wrote the `user` data source I think my examples weren't included because i wasnt naming the example file correctly - thus misunderstanding the default templates.

Given that, we no longer need the `user` template for user data source and we can mention templates as an option in the docs

### CODE REVIEW

**Comment by moduli** [comment]
This [PR](https://github.com/hashicorp/terraform-provider-boundary/pull/468) also added a `scope.md.tmpl`. Is that file still necessary?

**Comment by drewmullen** [comment]
Yes its still required bc it includes 1 extra line in the description

File: README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ If you're using vscode, this provider has delve debugging baked into the [.vscod
...
+This provider uses the [terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs/) tool to generate documentation. Each resource and data source requires a set of examples [examples](https://github.com/hashicorp/terraform-provider-boundary/tree/main/examples) then you execute a binary generate the docs which are placed in [./docs](https://github.com/hashicorp/terraform-provider-boundary/tree/main/docs). Optionally you can specify your own a [templates](https://github.com/hashicorp/terraform-provider-boundary/tree/main/templates) and for a resource or data source if you want to add custom information (example: [data "boundary_scope"](https://github.com/hashicorp/terraform-provider-boundary/blob/main/templates/data-sources/scope.md.tmpl#L11))
```
Comment (moduli) [inline README.md:76]:
```suggestion
This provider uses the [terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs/) tool to generate documentation. Each resource and data source requires a set of examples [examples](https://github.com/hashicorp/terraform-provider-boundary/tree/main/examples) then you execute a binary to generate the docs which are placed in [./docs](https://github.com/hashicorp/terraform-provider-boundary/tree/main/docs). Optionally, you can specify your own [templates](https://github.com/hashicorp/terraform-provider-boundary/tree/main/templates) for a resource or data source if you want to add custom information (example: [data "boundary_scope"](https://github.com/hashicorp/terraform-provider-boundary/blob/main/templates/data-sources/scope.md.tmpl#L11))
```

**Review by moduli — COMMENTED** [review]
Just some minor fixes on the README

**Comment by drewmullen** [comment]
@moduli committed your exaction recommendation

@psekar fyi

**Review by moduli — APPROVED** [review]

**Pull request merged**